pymc3/distributions/multivariate.py

```diff
@@ -690,6 +692,143 @@ def logp(self, x):
+                :math:`\sum x_i = n`
+    Mean        :math:`n \frac{a_i}{\sum{a_k}}`
+    ==========  ===========================================
+
+    Parameters
+    ----------
+    n : int or array
+        Total counts in each replicate. If n is an array its shape must be (N,)
+        with N = a.shape[0]
+
+    a : one- or two-dimensional array
+        Dirichlet parameter. Elements must be non-negative.
+        The number of categories is given by the length of the last axis.
+
+    shape : integer tuple
+        Describes shape of distribution. For example if n=array([5, 10]), and
+        p=array([1, 1, 1]), shape should be (2, 3).
+    """
+
+    def __init__(self, n, a, shape, *args, **kwargs):
+
+        super().__init__(shape=shape, defaults=("_defaultval",), *args, **kwargs)
+
+        if len(self.shape) > 1:
+            self.n = tt.shape_padright(n)
+            self.a = tt.as_tensor_variable(a) if np.ndim(a) > 1 else tt.shape_padleft(a)
+        else:
+            # n is a scalar, p is a 1d array
+            self.n = tt.as_tensor_variable(n)
+            self.a = tt.as_tensor_variable(a)
+
+        p = self.a / self.a.sum(-1, keepdims=True)
+
+        self.mean = self.n * p
+        # Mode is only an approximation. Exact computation requires a complex
+        # iterative algorithm as described in https://doi.org/10.1016/j.spl.2009.09.013
+        mode = tt.cast(tt.round(self.mean), "int32")
+        diff = self.n - tt.sum(mode, axis=-1, keepdims=True)
+        inc_bool_arr = tt.abs_(diff) > 0
+        mode = tt.inc_subtensor(mode[inc_bool_arr.nonzero()], diff[inc_bool_arr.nonzero()])
+        self._defaultval = mode
+
+    def _random(self, n, a, size=None, raw_size=None):
+        # numpy will cast dirichlet and multinomial samples to float64 by default
+        original_dtype = a.dtype
+
+        # Thanks to the default shape handling done in generate_values, the last
+        # axis of n is a dummy axis that allows it to broadcast well with `a`
+        n = np.broadcast_to(n, size)
+        a = np.broadcast_to(a, size)
+        n = n[..., 0]
+
+        # np.random.multinomial needs `n` to be a scalar int and `a` a
+        # sequence so we semi flatten them and iterate over them
+        n_ = n.reshape([-1])
+        a_ = a.reshape([-1, a.shape[-1]])
+        p_ = np.array([np.random.dirichlet(aa) for aa in a_])
+        samples = np.array([np.random.multinomial(nn, pp) for nn, pp in zip(n_, p_)])
+        samples = samples.reshape(a.shape)
+
+        # We cast back to the original dtype
+        return samples.astype(original_dtype)
+
+    def random(self, point=None, size=None):
+        """
+        Draw random values from Dirichlet-Multinomial distribution.
+
+        Parameters
+        ----------
+        point: dict, optional
+            Dict of variable values on which random values are to be
+            conditioned (uses default point if not specified).
```

Comment:
Can you add a very small example here, to make this kwarg clearer? Also, maybe using a more explicit name than `point` (if possible)?

Reply:
This signature and docstring is consistent with the other distributions in `pymc3.distributions`.

I think if we wanted to make a change like that we should probably do it for all of the distributions together, right?

Reply:
Good point. Then could you just add a small example of how `point` would be used? Since it's just in the docstring, no need to add that everywhere -- contrary to chaning the name of the kwarg for instance

Reply:
I think the point is used internally when doing posterior and prior predictive sampling. Not sure if users need to know about it in here. If anything it should be in the general docs I think.

Reply:
What do you mean by "general docs"? I agree users don't really care about that here; I'm thinking about developers -- to ease understanding and onboarding of future contributors

Reply:
Maybe update the `<Distribution>.random()` docstring for all distributions in a separate PR?

Reply:
Sounds good 👍 But then wouldn't adding this precision here already save time and provide an example for the subsequent PR?

Reply:
> What do you mean by "general docs"? I agree users don't really care about that here; I'm thinking about developers -- to ease understanding and onboarding of future contributors

Yes that's what I had in mind. It should be documented. It is just that here would be a very obscure place, since it actually applies to all the random methods. I agree with @bsmith89 suggestion

Reply:
Ok, that sounds like a good way forward so let's do that. Thanks for the discussion guys!

Reply:
Adding this to my list of follow-up issues/PRs :)